sdk/inc/azure/platform/az_mqtt5_pahoasync.h

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ *
+ * @brief Defines MQTT 5 constructs for Eclipse Paho Async.
+ *
+ * @note You MUST NOT use any symbols (macros, functions, structures, enums, etc.)
+ * prefixed with an underscore ('_') directly in your application code. These symbols
+ * are part of Azure SDK's internal implementation; we do not document these symbols
+ * and they are subject to change in future versions of the SDK which would break your code.
+ */
+
+#ifndef _az_MQTT5_PAHOASYNC_H
+#define _az_MQTT5_PAHOASYNC_H
+
+#include "MQTTAsync.h"
+#include <azure/core/internal/az_mqtt5_internal.h>
+
+#include <azure/core/_az_cfg_prefix.h>
+
+/**
+ * @brief MQTT 5 options for Paho Async.
+ *
+ */
+typedef struct
+{
+  /**
+   * @brief Platform options that are common across all MQTT 5 implementations.
+   */
+  az_mqtt5_options_common platform_options;
+
+  /**
+   * @brief The CA Trusted Roots path span interpretable by the underlying MQTT 5 implementation.
+   *
+   * @details We recommend avoiding configuring this option and instead using the default
+   * OpenSSL trusted roots instead.
+   */
+  az_span certificate_authority_trusted_roots;
+
+  /**
+   * @brief Whether to use TLS for the underlying MQTT 5 implementation.
+   */
+  bool disable_tls;
+} az_mqtt5_options;
+
+struct az_mqtt5
+{
+  struct
+  {
+    /**
+     * @brief Handle to the underlying MQTT 5 implementation (Paho Async).
+     */
+    MQTTAsync* pahoasync_handle;
+
+    /**
+     * @brief Platform MQTT 5 client that is common across all MQTT 5 implementations.
+     */
+    az_mqtt5_common platform_mqtt5;
+
+    /**
+     * @brief MQTT 5 options for Paho Async.
+     */
+    az_mqtt5_options options;
+  } _internal;
+};
+
+/**
+ * @brief MQTT5 property bag.
+ *
+ */
+typedef struct
+{
+  /**
+   * @brief Paho Async specific property array.
+   *
+   */
+  MQTTProperties* pahoasync_properties;
+
+  /**
+   * @brief Paho Async property structure used for constructing the property array.
+   *
+   */
+  MQTTProperty pahoasync_property;
```

Comment:
If I understand this correctly, this property struct is used as a temporary place to create a property before it's added to the bag, and then would be overwritten/reused the next time you create a property and add it to the bag. Is this correct? If so, can you add something like this in the comment so it's clear this shouldn't be used as a stable reference at any point?

Reply:
Correct, I could have added it to each of the property structs we have (`az_mqtt5_property_string_pair`) but we would have had to add property structs for byte, int and other primitive types. I clarified my comment!